src/jit/gentree.cpp

```diff
@@ -6246,15 +6246,310 @@ bool GenTree::TryGetUse(GenTree* def, GenTree*** use)
     assert(def != nullptr);
     assert(use != nullptr);
 
-    for (GenTree** useEdge : UseEdges())
+    switch (OperGet())
+    {
+        // Leaf nodes
```

Comment:
Yes--checking the operator kind has proven to be a slower approach in the past (e.g. we initially used kinds in the operand iterator and needed to change that to use a similar switch for efficiency reasons).